sycl/include/sycl/accessor.hpp

```diff
@@ -2200,29 +2203,29 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   iterator begin() const noexcept {
     return iterator::getBegin(
         get_pointer(),
-        detail::convertToArrayOfN<Dimensions, 1>(getMemoryRange()), get_range(),
-        get_offset());
+        detail::convertToArrayOfN<AdjustedDim, 1>(getMemoryRange()),
+        get_range<AdjustedDim>(), get_offset<AdjustedDim>());
```

Comment:
With my other suggested changes, you might need to make a private version that allows `get_range` and `get_offset` to be called with an alternative dimensionality.

Reply:
@steffenlarsen , thanks. I reverted  updates to get_range and get_offset and added private versions.